libraries/AP_RangeFinder/AP_RangeFinder_JRE_Serial.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_RangeFinder_JRE_Serial.h"
+
+#if AP_RANGEFINDER_JRE_SERIAL_ENABLED
+
+#include <GCS_MAVLink/GCS.h>
+
+#define FRAME_HEADER_1   0x52  // 'R'
+#define FRAME_HEADER_2   0x41  // 'A'
+
+#define DIST_MAX_CM 50000
+
+#define CCITT_INITPARA   0xffff
+#define CCITT_POLYNOMIAL 0x8408
+#define CCITT_OUTPARA    0xffff
+
+/* CRC16CCITT */
+static uint16_t CalcCRC16CCITT(uint8_t *cbuffer, uint16_t csize)
+{
+    uint16_t crc = CCITT_INITPARA;
+    for (int i = 0; i < csize; i++) {
+        crc ^= *cbuffer++;
+        for (int j = 0; j < 8; j++) {
+            if ((crc & 0x0001) != 0) {
+                crc >>= 1;
+                crc ^= CCITT_POLYNOMIAL;
+            } else {
+                crc >>= 1;
+            }
+        }
+    }
+    crc = crc ^ CCITT_OUTPARA;
+    return crc;
+}
+
+bool AP_RangeFinder_JRE_Serial::get_reading(float &reading_m)
```

Comment:
Can you describe the frame configuration as there does not seem to be a datasheet published.
BYTE 
0 : FRAME_HEADER_1 'R'
1: FRAME_HEADER_2 'A'
BYTE 0 : FRAME_HEADER_1 'R'
: :

Reply:
The data format is shown below.
![Fig1](https://github.com/ArduPilot/ardupilot/assets/113963003/f10d5a3d-f6ab-4c8e-b125-a73077ba411f)

Reply:
If possible I think we should add this packet structure into the comments at the top of the .cpp or .h file.  It's important for future developers to know the packet format in case they need to modify the code a bit.

Reply:
Since no datasheet has been published for this device, I will add more details to the header file.